banyand/liaison/grpc/grpc.go

```diff
@@ -52,6 +77,9 @@ func (s *shardInfo) Rev(message bus.Message) (resp bus.Message) {
 		s.log.Warn().Msg("invalid event data type")
 		return
 	}
+	s.RWMutex.Lock()
+	defer s.RWMutex.Unlock()
+	s.shardEvent = shardEvent
 	s.log.Info().
```

Comment:
* Add two separate `mutex` to protect `shardEvent` and `seriesEvent`. They don't want to block each other.
 * Pls parse the `action` of `shardEvent` and `seriesEvent` instead of accepting it directly.